src/V3Number.cpp

```diff
@@ -941,6 +950,50 @@ int V3Number::widthMin() const {
     return 1;  // one bit even if number is == 0
 }
 
+uint32_t V3Number::countBits(const V3Number& ctrl) const {
+    int n = 0;
+
+    switch (ctrl.bitIs(0)) {
+        case '0': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIs0(bit)) n++;
+            }
+            break;
+        }
+
+        case '1': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIs1(bit)) n++;
+            }
+            break;
+        }
+
+        case 'x': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIsX(bit)) n++;
+            }
+            break;
+        }
+
+        case 'z': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIsZ(bit)) n++;
+            }
+            break;
+        }
+    }
+    return n;
+}
+
+uint32_t V3Number::countBits(const V3Number& ctrl1, const V3Number& ctrl2,
+                             const V3Number& ctrl3) const {
+    int n = countBits(ctrl1);
+    if (ctrl2.bitIs(0) != ctrl1.bitIs(0)) n += countBits(ctrl2);
+    if ((ctrl3.bitIs(0) != ctrl1.bitIs(0)) && (ctrl3.bitIs(0) != ctrl2.bitIs(0)))
+        n += countBits(ctrl3);  // TODO check only bit 0
```

Comment:
```suggestion
    if ((ctrl3.bitIs(0) != ctrl1.bitIs(0)) && (ctrl3.bitIs(0) != ctrl2.bitIs(0))) {
        n += countBits(ctrl3);  // TODO check only bit 0
    }
```

Reply:
Thanks, I'll also add a patch to remove the TODO comment